services/event/service/event_service.go

```diff
@@ -286,3 +298,22 @@ func MarkUserAsAttendingEvent(event_name string, user_id string) error {
 
 	return err
 }
+
+const CheckInTimeIntervalInSeconds = 1200
+
+/*
+	Check if an event is active, i.e., that check-ins are allowed for the event at the current time.
+	Returns true if and only if abs(Current Time - Event Start Time) <= 1200 seconds (20 minutes).
+*/
+func IsEventActive(event_name string) (bool, error) {
+	event, err := GetEvent(event_name)
+
+	if err != nil {
+		return false, err
+	}
+
+	startTime := event.StartTime
+	t := time.Now().Unix()
+
+	return math.Abs((float64)(t-startTime)) <= CheckInTimeIntervalInSeconds, nil
```

Comment:
We might have some long events (1+ hours). I'd say people should be able to checkin to the event up until the end of the event. We can probably also shorten up the pre-event checkin time. I'd say 15 minutes before an event till the end of the event is the time frame we want.